### PR TITLE
Mip family id compatibility

### PR DESCRIPTION
**Pull request by henrikstranneheim**
This PR adds/fixes:

- Removes unused keys from parsing mip files that was previously used to store in HK
- Adds subs to get `recipe`key or fall back on `program`
- Added logic to use `case_id` or fallback on `family_id`
- Removed left-over traiblazer check defs

### How to prepare for test
- [x] ssh to hasta
- [x] activate stage: `us`
- [x] request trailblazer-stage on hasta: `paxa`
- [x] install on stage:
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-trailblazer-stage.sh mip_family_id_compatibility`

### How to test
- [x] login to hasta
- [x] do `us`
1. `trailblazer scan`
1. `trailblazer ls`
1. `trailblazer cancel -j 219398`
1. `cg upload delivery-report -f vitalmouse`
1. `cg upload delivery-report -f strongbison`
1. `cg upload scout vitalmouse`

### Expected test outcome
1. `trailblazer scan` ✅ 
1. `trailblazer ls` ✅ 
1. `trailblazer cancel -j 219398` ✅ 
1. The delivery-report should be created without any missing values from the MIP analysis .i.e. duplicates, genome_build, analysis_sex ✅ 
1. The delivery-report should be created without any missing values from the MIP analysis .i.e. duplicates, genome_build, analysis_sex ✅ 
1. The case should successfully be uploaded to Scout. The rank models, analysis date and peddy info should be present in Scout. ✅ 

## Review
- [x] code approved by PG
- [x] tests executed by HS
- [x] "Merge and deploy" approved by PG
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by henrikstranneheim** [comment]
### How to test
- [ ] login to hasta
- [ ] do `us`
1. trailblazer scan
Screenshot 2020-08-20 at 14 56 47

1. trailblazer ls
Screenshot 2020-08-20 at 14 57 25

1. trailblazer cancel -j 219398
Screenshot 2020-08-20 at 14 58 43

1. `cg upload delivery-report -f vitalmouse`
Screenshot 2020-08-20 at 14 53 55

1. `cg upload delivery-report -f strongbison`
Screenshot 2020-08-20 at 14 55 38

File: trailblazer/mip/files.py [hunk 1]
```diff
@@ -17,7 +15,7 @@ def parse_config(data: dict) -> dict:
     """
     return {
         'email': data.get('email'),
-        'case': data['case_id'],
+        'case': data['case_id'] if 'case_id' in data else data['family_id'],
```
Comment (patrikgrenfeldt) [inline trailblazer/mip/files.py:18]:
```suggestion
        'case': data['case_id'] if 'case_id' in data else data['family_id'],  # family_id for MIP<7
```

File: trailblazer/mip/files.py [hunk 2]
```diff
@@ -74,73 +78,41 @@ def parse_sampleinfo(data: dict) -> dict:
...
 
     return outdata
 
+def get_plink_samples(metrics: dict) -> dict:
+    """Get plink samples"""
+    plink_sexcheck = None
+    plink_samples = {}
+    if 'recipe' in metrics:
+        plink_sexcheck = metrics['recipe'].get('plink_sexcheck', {}).get('sample_sexcheck')
+    elif 'program' in metrics:
```
Comment (patrikgrenfeldt) [inline trailblazer/mip/files.py:106]:
```suggestion
    elif 'program' in metrics:    # for MIP<7
```

File: trailblazer/mip/files.py [hunk 3]
```diff
@@ -62,6 +60,12 @@ def get_sampleinfo_date(data: dict) -> str:
 
     return data['analysis_date']
 
+def get_rank_model_version(sample_info: dict, rank_model_type: str, step: str) -> str:
+    """Get rank model version"""
+    if 'recipe' in sample_info:
+        return sample_info['recipe'][step][rank_model_type]['version']
+    if 'program' in sample_info:
```
Comment (patrikgrenfeldt) [inline trailblazer/mip/files.py:67]:
```suggestion
    elif 'program' in sample_info:    # for MIP<7
```

File: .gitignore [hunk 4]
```diff
@@ -49,6 +49,7 @@ TRASH*
 _DEVELOP/
 .vscode/
 
+
 # Development
```
Comment (jemten) [inline .gitignore:53]:
```suggestion
```

**Review by jemten — APPROVED** [review]
Looks great!

File: .gitignore (hunk 4 above)
Comment (jemten) [reply .gitignore:53]: Never mind

**Comment by henrikstranneheim** [comment]
Deployed on hasta:
Screenshot 2020-08-26 at 16 01 00

**Pull request merged**